docs/template_plugin/tests/functional/op_reference/grn.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_core.hpp>
+#include <ie_ngraph_utils.hpp>
+#include <ngraph/ngraph.hpp>
+#include <shared_test_classes/base/layer_test_utils.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+
+struct GrnParams {
+    template <class IT, class OT>
+    GrnParams(const float bias, const ngraph::PartialShape& shape, const ngraph::element::Type& iType, const ngraph::element::Type& oType,
+              const std::vector<IT>& iValues, const std::vector<OT>& oValues, size_t iSize = 0, size_t oSize = 0)
+        : bias(bias), pshape(shape), inType(iType), outType(oType), inputData(CreateBlob(iType, iValues, iSize)), refData(CreateBlob(oType, oValues, oSize)) {}
+    float bias;
+    ngraph::PartialShape pshape;
+    ngraph::element::Type inType;
+    ngraph::element::Type outType;
+    InferenceEngine::Blob::Ptr inputData;
+    InferenceEngine::Blob::Ptr refData;
+};
+
+class ReferenceGrnLayerTest : public testing::TestWithParam<GrnParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.bias, params.pshape, params.inType);
+        inputData = {params.inputData};
+        refOutData = {params.refData};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<GrnParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "bias=" << param.bias << "_";
+        result << "shape=" << param.pshape << "_";
+        result << "iType=" << param.inType << "_";
+        result << "oType=" << param.outType;
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(float bias, const PartialShape& input_shape, const element::Type& input_type) {
+        const auto in = std::make_shared<op::Parameter>(input_type, input_shape);
+        const auto grn = std::make_shared<op::v0::GRN>(in, bias);
+        return std::make_shared<Function>(NodeVector {grn}, ParameterVector {in});
+    }
+};
+
+TEST_P(ReferenceGrnLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_GRN_With_Hardcoded_Refs, ReferenceGrnLayerTest,
+    ::testing::Values(
+        // bias 1e-6 // 2D // 3D // 4D
+        GrnParams(1e-6, ngraph::PartialShape {3, 4}, ngraph::element::f32, ngraph::element::f32, std::vector<float> {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12},
```

Comment:
Should we extend these tests for other data types?

Reply:
Refactored to generate params with different types (bf16, f16, f32), please take a look.